defrag/modules/docs.py

```diff
@@ -0,0 +1,163 @@
+    return built
+
+
+def sorted_on_score(items: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    return sorted(items, key=lambda x: x["score"], reverse=True)
+
+
+def search_index(idx: Index, source: str, keywords: str) -> List[Dict[str, Any]]:
+    return list(map(lambda item: thin_prime(source, item), idx.search(keywords)))
+
+
+def set_global_index(source: str, idx: Index) -> None:
+    global indexes
+    indexes[source]["index"] = idx
+
+
+def ready_to_index(sources: List[str]) -> bool:
+    for s in sources:
+        if not indexes[s]["index"]:
+            return False
+    return True
+
+
+def make_index_search_leap(leap_data: str, keywords: str) -> Tuple[Index, List[Dict[str, Any]]]:
+    idx = make_leap_index(leap_data)
+    return idx, search_index(idx, "leap", keywords)
+
+
+def make_index_search_tumbleweed(tw_data: bytes, keywords: str) -> Tuple[Index, List[Dict[str, Any]]]:
+    idx = make_tumbleweed_index(tw_data)
+    return idx, search_index(idx, "tumbleweed", keywords)
+
+
+async def make_search_set_indexes_in_parallel(keywords: str) -> List[Dict[str, Any]]:
+
+    leap, tw = await asyncio.gather(get_data("leap"), get_data("tumbleweed"))
+    results = []
+    sys.setrecursionlimit(0x100000)
+    with ProcessPoolExecutor(max_workers=2) as executor:
+        leap_worker = executor.submit(
+            make_index_search_leap, **{"leap_data": leap, "keywords": keywords})
+        tw_worker = executor.submit(
+            make_index_search_tumbleweed, **{"tw_data": tw, "keywords": keywords})
+        leap_index, leap_results = leap_worker.result()
+        tw_index, tw_results = tw_worker.result()
+        set_global_index("leap", leap_index)
+        set_global_index("tumbleweed", tw_index)
+        results = leap_results + tw_results
+    return results
+
+
+def search_indexes_in_parallel(keywords: str) -> List[Dict[str, Any]]:
+    with ProcessPoolExecutor(max_workers=2) as executor:
+        leap_worker = executor.submit(
+            search_index, **{"idx": indexes["leap"]["index"], "source": "leap", "keywords": keywords})
+        tw_worker = executor.submit(
+            search_index, **{"idx": indexes["tumbleweed"]["index"], "source": "tumbleweed", "keywords": keywords})
+        results = leap_worker.result() + tw_worker.result()
+    return results
+
+
+@app.get("/" + __MOD_NAME__ + "/single/{source}/{keywords}")
+async def search_single_source_docs(source: str, keywords: str) -> QueryResponse:
+    if not ready_to_index([source]):
+        if source == "tumbleweed":
+            set_global_index("tumbleweed", make_tumbleweed_index(await get_data(source)))
+        else:
+            set_global_index("leap", make_leap_index(await get_data(source)))
+    results = sorted_on_score(search_index(
+        indexes[source]["index"], source, keywords))
+    return QueryResponse(query=Query(service="search_docs"), results_count=len(results), results=results)
+
+
+@app.get("/" + __MOD_NAME__ + "/merged/{keywords}")
```

Comment:
```suggestion
@app.get("/" + __MOD_NAME__ + "/merged/")
```
This would make it so that we take the search term as a query parameter, which I think is nicer.